### DISCUSSION TITLE
Workflow for ONT Data

### DISCUSSION THREAD

**Discussion by marchoeppner** [bio-raum/FooDMe2#34] [General]
I am not 100% sure this will work - there are some comments in the Dada2 github issues regarding the theoretical compatibility of ONT reads with >Q20, but nothing substantial has been documented. 

Worth looking into. 

I also checked how the ONT software suite (epi2me) is handling amplicons. The short story is that it is a bit of a pain to read, even though their code is technically "nextflow". From what I can tell they use a combination of de-novo assembly and subsequent error correction. Just going by their parameters and options, it looks a bit dodgy [code](https://github.com/epi2me-labs/wf-amplicon/blob/master/modules/local/de-novo.nf)

Basically, when I do this with our ONT test data, if get one assembled sequence that could be sheep (one of the primary components of that test sample). And after error correction, it comes back as Horse ;) So..room for improvement..

**Comment by marchoeppner** [comment]
I have implemented a very basic ONT workflow now (--ont, and optionally --ont_min_qual <int>), for Vsearch (and Dada2 also accepts the reads). But in either case, the results look a bit dodgy...

Basically, we need high quality reads and the test data we have right now does not really qualify. The amount of reads @ Q20 is quite low, and anything below is probably too noisy (or will need a lot of time to develop a suitable processing chain). You can play with that aspect by setting different options for --ont_min_q (default = 15). 

Maybe you have an idea how to approach this (short of getting better data ;) )

**Reply by gregdenay** [reply]
I could look for other datasets, I have candidates, maybe the Q20 would be better.

**Comment by marchoeppner** [comment]
I have updated the workflow for ONT a bit to include a step to orient the trimmed reads against the chosen reference database (FASTA). This has a the minor issue that we do not (and cannot) have the matching fasta file for the larger databases. So this currently only works for Midori, Unite and Refseq mito. A workaround might be to simply use Refseq Mito, since this database should presumably contain the relevant targets. But if we ever decided to also support non-mitochondrial sequences, this obviously wouldn't work. 

Basically, the pipeline will align the individual reads against the database and try to return them all in a consistent orientation. Doing that seems to help recover more and larger OTUs/clusters, although the issue of the high error rate persists and we only get 2-3 useable clusters before the noise becomes too high for our BLAST filters to tolerate. But it's a step in the right direction, I reckon.

**Comment by marchoeppner** [comment]
As a small addendum - since I was a little curious about the most likely actual composition of our ONT test data, I simply blasted each read against RefSeq Mito, no OTU clustering. Turns out, the OTU bases results are actually more or less spot on. Probably worth discussing at the next meeting.
